### PR TITLE
get all api for supplier category

### PR DESCRIPTION
**Pull request by Keerthana65**
get all API - Supplier Category

### CODE REVIEW

File: tokyo-supermix-data-common/src/main/java/com/tokyo/supermix/data/dto/SupplierCategoryDto.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package com.tokyo.supermix.data.dto;
+
+public class SupplierCategoryDto {
+
+	private Long id;
+	private String category;
```
Comment (SSuja) [inline tokyo-supermix-data-common/src/main/java/com/tokyo/supermix/data/dto/SupplierCategoryDto.java]: Add validation

File: tokyo-supermix-server/src/main/java/com/tokyo/supermix/server/services/SupplierCategoryServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import com.tokyo.supermix.data.repositories.SupplierCategoryRepository;
+
+@Service
+public class SupplierCategoryServiceImpl implements SupplierCategoryService {
+	@Autowired
+	private SupplierCategoryRepository supplierCategoryRepository;
+
+	@Override
+	public List<SupplierCategory> getAllSupplierCategories() {
+		return supplierCategoryRepository.findAll();
```
Comment (SSuja) [inline tokyo-supermix-server/src/main/java/com/tokyo/supermix/server/services/SupplierCategoryServiceImpl.java]: @Transactional Missing

File: tokyo-supermix-server/src/main/java/com/tokyo/supermix/server/controller/SupplierCategoryController.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+	@GetMapping(value = EndpointURI.GET_ALL_SUPPLIER_CATEGORY)
+	public ResponseEntity<Object> getAllSupplierCategories() {
+		List<SupplierCategory> supplierCategoryList = supplierCategoryService.getAllSupplierCategories();
+		if (supplierCategoryList.isEmpty()) {
+			logger.debug("Supplier Category is empty");
+			return new ResponseEntity<>(
+					new ApiResponse(RestApiResponseStatus.VALIDATION_FAILURE, ValidationConstance.NO_DATA_FOUND),
+					HttpStatus.BAD_REQUEST);
+		}
+		return new ResponseEntity<>(mapper.map(supplierCategoryList, SupplierCategoryDto.class), HttpStatus.OK);
```
Comment (SSuja) [inline tokyo-supermix-server/src/main/java/com/tokyo/supermix/server/controller/SupplierCategoryController.java]: Use "ListConententgRepspone class"

**Comment by Keerthana65** [comment]
> GET_ALL_SUPPLIER_CATEGORY does not exist EndpointURI and change this name also as SUPPLIERS

This task is resolved

**Pull request closed without merging**